src/main/kotlin/frc/robot/subsystems/ExampleSubsystem.kt

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+package frc.robot.subsystems
+
+import edu.wpi.first.wpilibj.DigitalInput
+import edu.wpi.first.wpilibj.motorcontrol.PWMSparkMax
+import edu.wpi.first.wpilibj2.command.Command
+import edu.wpi.first.wpilibj2.command.SubsystemBase
+
+/** Creates a new ExampleSubsystem.  */
+class ExampleSubsystem(
+    exampleConstructorArg: Double,
+    /**
+     * An example of a property that is also a constructor argument.
+     * The value of inputID depends on the value passed in
+     * when the ExampleSubsystem is created.
+     */
+    private val digitalInputID: Int,
+    private val motorID: Int,
+): SubsystemBase() {
+    init {
+        // This is run when the ExampleSubsystem instance is created.
+    }
+
+    /**
+     * Example command factory method.
+     *
+     * @return a command
+     */
+    fun exampleMethodCommand(): Command =
+        // Inline construction of command goes here.
+        // Subsystem::runOnce implicitly requires `this` subsystem.
+        runOnce {
+            println("This is an example command.")
+        }
+
+    /**
+     * An example of a regular kotlin property
+     * whose value cannot be changed.
+     */
+    private val digitalInput = DigitalInput(digitalInputID)
+    private val motor = PWMSparkMax(motorID)
+
+    /**
+     * An example of a regular kotlin property
+     * that can be modified.
+     */
+    private var exampleMutableProperty = 2.0
```

Comment:
I don't think this class needs to be examples of kotlin features. I think for this repo we can assume familiarity with the language